identity-apps-core/apps/recovery-portal/src/main/webapp/includes/language-switcher.jsp

```diff
@@ -100,6 +101,15 @@
         document.cookie = name + "=" + (value || "")  + expires + domain + "; path=/";
     }
 
+    /**
+     * Handles language change by setting the `ui_locale` cookie from the url params ui_locale.
+     */
+     function setCookieFromUrlParamUiLocale(computedLocale) {
```

Comment:
The function name implies it retrieves the cookie value from the url params and sets the cookie, but what it actually does is setting the cookie with whatever value passed to it. The function body doesn't have anything related to url params.

Therefore let's rename the function as follows,

```suggestion
     function setUILocaleCookie(locale) {
```

and use this function here:

https://github.com/wso2/identity-apps/blob/master/identity-apps-core/apps/recovery-portal/src/main/webapp/includes/language-switcher.jsp#L109#L109-111

That would be much better in terms of readability.

Please update the other files as well. TIA!

Reply:
resolved with e8975caba6a056f36e0fe07faedc223ef8cd0719